File: 1476-count-negative-numbers-in-a-sorted-matrix/1476-count-negative-numbers-in-a-sorted-matrix.cpp
class Solution {
public:

    int getNegativesCount(vector<int>& row) {

        int n = row.size();
        int left = 0;
        int right = n-1;

        while(left<=right){
            int mid = left + (right - left)/2;

            if(row[mid]>=0){
                left = mid + 1;
            }else{
                right = mid - 1;
            }
        }

        if(left >= n){
            return 0;
        }else{
            return n-left;
        }
        
    }

    int countNegatives(vector<vector<int>>& grid) {
        
        int count = 0;

        for(int i=0; i<grid.size(); i++){
            count += getNegativesCount(grid[i]);
        }

        return count;

    }
};

File: 1476-count-negative-numbers-in-a-sorted-matrix/1476-count-negative-numbers-in-a-sorted-matrix.py
class Solution:

    def getNegativesCount(self, row: List[int]) -> int:

        n = len(row)
        left, right = 0, n-1

        while left <= right:
            mid = left + (right-left)//2

            if row[mid] >= 0:
                left = mid + 1
            else:
                right = mid - 1

        if left >= n:
            return 0
        return n-left

    def countNegatives(self, grid: List[List[int]]) -> int:
        count = 0

        for row in grid:
            count += self.getNegativesCount(row)

        return count